JitsiConference.js

```diff
@@ -1623,7 +1795,519 @@ JitsiConference.prototype.broadcastEndpointMessage = function(payload) {
 };
 
 JitsiConference.prototype.isConnectionInterrupted = function() {
-    return this.connectionIsInterrupted;
+    return this.p2pEstablished
+        ? this.isP2PConnectionInterrupted : this.isJvbConnectionIsInterrupted;
+};
+
+/**
+ * Handles {@link XMPPEvents.CONNECTION_INTERRUPTED}
+ * @param {JingleSessionPC} session
+ * @private
+ */
+JitsiConference.prototype._onIceConnectionInterrupted = function(session) {
+    if (session.isP2P) {
+        this.isP2PConnectionInterrupted = true;
+    } else {
+        this.isJvbConnectionIsInterrupted = true;
+    }
+    if (session.isP2P === this.p2pEstablished) {
+        this.eventEmitter.emit(JitsiConferenceEvents.CONNECTION_INTERRUPTED);
+    }
+};
+
+/**
+ * Handles {@link XMPPEvents.CONNECTION_ICE_FAILED}
+ * @param {JingleSessionPC} session
+ * @private
+ */
+JitsiConference.prototype._onIceConnectionFailed = function(session) {
+    if (session.isP2P) {
+        this._stopP2PSession('connectivity-error', 'ICE FAILED');
+    }
+};
+
+/**
+ * Handles {@link XMPPEvents.CONNECTION_RESTORED}
+ * @param {JingleSessionPC} session
+ * @private
+ */
+JitsiConference.prototype._onIceConnectionRestored = function(session) {
+    if (session.isP2P) {
+        this.isP2PConnectionInterrupted = false;
+    } else {
+        this.isJvbConnectionIsInterrupted = false;
+    }
+
+    if (session.isP2P === this.p2pEstablished) {
+        this.eventEmitter.emit(JitsiConferenceEvents.CONNECTION_RESTORED);
+    }
+};
+
+/**
+ * Accept incoming P2P Jingle call.
+ * @param {JingleSessionPC} jingleSession the session instance
+ * @param {jQuery} jingleOffer a jQuery selector pointing to 'jingle' IQ element
+ * @private
+ */
+JitsiConference.prototype._acceptP2PIncomingCall
+= function(jingleSession, jingleOffer) {
+    jingleSession.setSSRCOwnerJid(this.room.myroomjid);
+
+    this.isP2PConnectionInterrupted = false;
+
+    // Accept the offer
+    this.p2pJingleSession = jingleSession;
+
+    this.p2pJingleSession.initialize(
+        false /* initiator */, this.room, this.rtc);
+
+    const localTracks = this.getLocalTracks();
+
+    logger.debug(`Adding ${localTracks} to P2P...`);
+    this.p2pJingleSession.addLocalTracks(localTracks).then(
```

Comment:
I don't get what you mean here. Why would you want to catch ? Can you be more specific ? What code needs to be executed if something fails ?

There is reject handler already that prints an error, no ?